platform/mds/mds-web/src/main/resources/webapp/js/directives.js

```diff
@@ -2409,6 +2409,59 @@
         };
     });
 
+    directives.directive('floatValidity', function() {
+        var FLOAT_REGEXP = new RegExp('^[-]?\\d+(\\.\\d+)?$'),
+        TWOZERO_REGEXP = new RegExp('^[-]?0+\\d+(\\.\\d+)?$');
+        return {
+            require: 'ngModel',
+            link: function(scope, element, attrs, ctrl) {
+                var elm = angular.element(element), originalValue;
+                ctrl.$parsers.unshift(function(viewValue) {
+                    if (viewValue === '' || FLOAT_REGEXP.test(viewValue)) {
+                        // it is valid
+                        ctrl.$setValidity('float', true);
+                        originalValue = viewValue;
+                        viewValue = parseFloat(viewValue);
+                        if (isNaN(viewValue)) {
+                            viewValue = '';
+                        }
+                        if (TWOZERO_REGEXP.test(originalValue)) {
+                            setTimeout(function () {
+                                elm.val(viewValue);
+                            }, 1000);
```

Comment:
If user types more than one zero it makes it one zero after one second. It is already used for 'decimalValidity'.